server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutes.java

```diff
@@ -52,9 +61,42 @@
 @Produces(Constants.JSON_CONTENT_TYPE)
 public class DeletedMessagesVaultRoutes implements Routes {
 
+    enum UserVaultAction {
+        RESTORE("restore");
+
+        static Optional<UserVaultAction> getAction(String value) {
+            Preconditions.checkArgument(StringUtils.isNotBlank(value), "action cannot be null, empty or blank");
+
+            return Stream.of(values())
+                .filter(action -> action.value.equals(value))
+                .findFirst();
+        }
+
+        static String joinedPlainValues() {
+            return Joiner.on(",").join(UserVaultAction.plainValues());
+        }
+
+        private static List<String> plainValues() {
```

Comment:
enum has a built-in method `values` that does exactly that. Why would you need a list at all?

Reply:
I don't need a List of Enum, I need a list of values of properties in side the Enum

Reply:
ok

Reply:
ok